### ISSUE TITLE
Allow specifying hostname to use when running `bun dev`

### ISSUE THREAD

**Issue by CompuIves** [oven-sh/bun#820]
### What is the problem this feature will solve?

I'm trying to run the React example `bun dev` on a VM running in the cloud on CodeSandbox [here](https://codesandbox.io/p/github/codesandbox/codesandbox-template-bun/draft/small-snowflake), (you can see the running dev server here: https://sidvt7-3000.preview.csb.app/) but I've noticed that the requests fail because they go directly to the IP of the node of the VM:

<img width="1172" alt="image" src="https://user-images.githubusercontent.com/587016/180278076-08829100-b728-4640-bc26-2cb20dc7289f.png">

I would like the requests to go to either `https://sidvt7-3000.preview.csb.app/` or `http://localhost:3000` (we proxy localhost:3000 requests using a service worker). Is there a way that we can specify in a config or something what the hostname should be that Bun rewrites URLs to?

### What is the feature you are proposing to solve the problem?

A way to define a HOST (maybe as env var) that Bun can use for rewriting its URL. That would allow me to create a runnable example of Bun on CodeSandbox and play with it more.

### What alternatives have you considered?

We could install a service worker that proxies the request from the IP to relative URL, but that feels like a hacky solution to this problem.

By the way, the default http example _does_ work 😄

**Comment by Jarred-Sumner** [comment]
Try passing `--origin https://my-absolute-url.preview.csb.app` to `bun dev`

If you run into further issues, try also passing `--port` but you shouldn't have to

**Comment by CompuIves** [comment]
Thanks for the quick reply! I've tried adding `--origin` to `bun dev`. Tried both this:
```
bun dev --origin https://sidvt7-3000.preview.csb.app --port 3000
```
And this
```
bun dev --origin http://localhost:3000 --port 3000
```

Also with out `--port`. However, the requests seem to be unchanged. I was thinking, could it maybe be related to a cache or something? The request to `/src/index.jsx` (referenced from `index.html`) is correct, but the file itself contains import links with the absolute IP:

<img width="1373" alt="image" src="https://user-images.githubusercontent.com/587016/180413512-1b89b5a1-009e-49b4-aa0e-078a3fda4704.png">

Tested with `v1.0.2` and `v1.0.4`.
